hazelcast/src/main/java/com/hazelcast/spi/impl/operationparker/impl/WaitSetEntry.java

```diff
@@ -210,4 +225,23 @@ protected void toString(StringBuilder sb) {
         sb.append(", expirationTimeMs=").append(expirationTimeMs);
         sb.append(", valid=").append(valid);
     }
+
+    @Override
+    public void sendResponse(WaitSetEntry op, Object response) {
+        // WaitSetEntry operation should never send a normal (non-exceptional) response.
+        // It's because WaitSetEntry operation is only executed to notify the inner BlockingOperation about
+        // expiration/timeout/cancellation. Upon the notification the inner BlockingOperation will use its
+        // own logic and its own response handler to react on the event.
+
+        // WaitSetEntry operation may still receive an exceptional response. For example when there is a partition
+        // migration in progress. This is not a big deal as the OperationParker will eventually retry it anyway.
+        if (response instanceof Throwable) {
+            // we log on finest only, as it's meant for debugging only. See #logError for regular
+            // user-facing logging on errors.
+            getLogger().finest("Error while executing " + this, (Throwable) response);
+        } else {
+            getLogger().warning("Error while executing " + this + " and the response is NOT Throwable. This is "
```

Comment:
we could do something like: `assert false "A non-exceptional response set to WaitSetEntry"`
wdyt?